### ISSUE TITLE
Feature Request: repo name in console title

### ISSUE THREAD

**Issue by daniel-liuzzi** [JanDeDobbeleer/oh-my-posh#655]
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] I looked for duplicate issues before submitting this one

### Description

Before oh-my-posh, I used to set my posh-git prompt as follows:

```PowerShell
$GitPromptSettings.WindowTitle = {
  param($GitStatus)
  if ($GitStatus) { $GitStatus.RepoName } else { Get-PromptPath }
}
```

It set the console title to `(Get-GitStatus).RepoName` if I'm in a Git repository, or to `pwd` otherwise. This worked beautifully for me, as I'm in a repo 99% of the time, and I could quickly glance the tab title to see what project I'm in.¹

I couldn't find a way to replicate this in oh-my-posh. I checked the console title settings in the docs, but there doesn't seem to be anything for Git. I also tried `"console_title": false` to have posh-git "take over" to no avail; as soon as OMP loads `$GitPromptSettings.WindowTitle` dies.

Would it be possible to have something like `"console_title_style": "repo"`? or better yet, a new console-title-template variable? Or at the very least, a way to have oh-my-posh "leave the title alone" so that posh-git can be used for that?

Awesome project BTW!

### Environment

- Oh my Posh version: 3.130.0
- Theme: own
- Operating System: Windows 10 20H2
- Shell: PowerShell 7.1.3
- Terminal: Windows Terminal Preview

---

¹I know I can use the folder name, but this doesn't work nearly as well. For example, every repo at my day job enforces a `src/` directory where code goes into. This would result in a "src" title since we're barely ever at the root. This gets worse when we have multiple projects open (bunch of "src" tabs, can't tell which is which.)

**Comment by JanDeDobbeleer** [comment]
I gave this some thought but decided against adding the functionality explicitly as it opens Pandora's box a bit for the title. That said, there is a way to have this with the current functionality. We have a function you can override to do basically anything before the prompt is rendered which is mainly used to populate environment variables so that oh-my-posh can use it.

Adding the following in your `$PROFILE` **after** you import or invoke oh-my-posh will set up everything correctly:

```powershell
function Set-EnvVars {
    Write-Host $GitStatus
    if (-not $GitStatus) {
        $env:TITLE = Get-PromptPath
        return
    }
    $env:TITLE = $GitStatus.RepoName
}
New-Alias -Name 'Set-PoshContext' -Value 'Set-EnvVars' -Scope Global -Force
```

You can then access the `TITLE` env var from oh-my-posh in the console title template:

```json
"console_title": true,
"console_title_style": "template",
"console_title_template": "{{ .Env.TITLE }}"
```

And you have the same effect. I did notice we invoke posh-git too late so I created a fix for that in #658

**Comment by daniel-liuzzi** [comment]
That worked a treat, thank you! I get your point about Pandora's box and agree. If every time somebody like myself requests a new variable, you add it, it won't be long before templates turn into everything but the kitchen sink.

Thanks again for prompt response 👍  (no pun intended)
